PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-lib/src/main/java/com/microsoft/azure/toolkit/intellij/common/messager/IntellijAzureMessage.java

```diff
@@ -0,0 +1,238 @@
+                .map(IntellijAzureMessage::transformURLIntoLinks)
+                .map(c -> String.format(", because <span style='font-style: italic;'>%s</span>", c))
+                .orElse("");
+        final String action = Optional.of(throwable)
+                .filter(t -> t instanceof AzureToolkitRuntimeException)
+                .map(t -> ((AzureToolkitRuntimeException) t).getAction())
+                .map(IntellijAzureMessage::transformURLIntoLinks)
+                .map(c -> String.format("<p>%s</p>", c))
+                .orElse("");
+        if (includingDetails) {
+            final String details = this.getDetails(operations);
+            final String style = "margin:0;margin-top:2px;padding-left:0;list-style-type:none;";
+            final String detailsMsg = StringUtils.isNotBlank(details) ? String.format("<div>Call Stack:</div><ul style='%s'>%s</ul>", style, details) : "";
+            return "<html>" + failure + cause + action + detailsMsg + "</html>";
+        }
+        return "<html>" + failure + cause + action + "</html>";
+    }
+
+    public String getDetails() {
+        return this.getDetails(this.getOperations());
+    }
+
+    private String getDetails(List<? extends IAzureOperation> operations) {
+        return operations.size() < 2 ? "" : operations.stream()
+                .map(IAzureOperation::getTitle)
+                .map(title -> String.format("<li>&#9679; %s</li>", StringUtils.capitalize(title.toString())))
+                .collect(Collectors.joining(""));
+    }
+
+    @Nonnull
+    public List<AnAction> getAnActions() {
+        return Arrays.stream(this.getActions()).map(a -> toAction(a, original)).collect(Collectors.toList());
+    }
+
+    @Nullable
+    public Boolean getBackgrounded() {
+        return Optional.ofNullable(this.backgrounded)
+                .or(() -> Optional.ofNullable(AzureTaskContext.current().getTask())
+                        .map(AzureTask::getBackgrounded)).orElse(null);
+    }
+
+    @Nullable
+    private static String getCause(@Nonnull Throwable throwable) {
+        final Throwable root = getRecognizableCause(throwable);
+        if (Objects.isNull(root)) {
+            return null;
+        }
+        String cause = null;
+        if (root instanceof ManagementException) {
+            cause = ((ManagementException) root).getValue().getMessage();
+        } else if (root instanceof HttpResponseException) {
+            cause = ((HttpResponseException) root).getResponse().getBodyAsString().block();
+        }
+        return StringUtils.firstNonBlank(cause, root.getMessage());
+    }
+
+    @Nullable
+    private static Throwable getRecognizableCause(@Nonnull Throwable throwable) {
+        final List<Throwable> throwables = ExceptionUtils.getThrowableList(throwable);
+        for (int i = throwables.size() - 1; i >= 0; i--) {
+            final Throwable root = throwables.get(i);
+            if (root instanceof AzureOperationException) {
```

Comment:
is there any cases that `AzureOperationException` is the root exception? Seems that it will be a system issue if `AzureOperationException` is the root cause